prover/src/verify.rs

```diff
@@ -218,12 +222,44 @@ impl SP1Prover {
         // Verify the proof with the corresponding public inputs.
         prover.verify(proof, &vkey_hash, &committed_values_digest, build_dir);
 
-        // Verify that the vk hash of the SP1VerifyingKey matches the vk hash specified in the proof.
-        let vk_hash = vk.hash_bn254().as_canonical_biguint();
-        if vk_hash != vkey_hash {
-            return Err(anyhow::Error::msg("vk hash mismatch"));
-        }
-
         Ok(())
     }
 }
+
+/// Verify the vk_hash and public_values_hash in the public inputs of the Groth16Proof match the expected values.
```

Comment:
I think it might make more sense to implement this code directly on proof as a impl function on self.

Generally speaking, it seems like all of these functions are always going to be called against the object.